src/commands/publish.ts

```diff
@@ -0,0 +1,77 @@
+import * as path from "path";
+import cli from "cli-ux";
+
+import TartCommand from "../TartCommand";
+import { createExecaCommand, parseGitOutput, GIT_FLAGS } from "../utils";
+
+export default class Publish extends TartCommand {
+  static description = "publish specified dump file to a remote repository";
+
+  static examples = [`$ tart publish filename`];
+
+  static flags = {
+    ...TartCommand.flags,
+  };
+
+  static args = [
+    {
+      name: "file",
+      required: true,
+      description: "name of the file",
+    },
+  ];
+
+  async run() {
+    await this.runHook("beforePublish");
+
+    const { args } = this.parse(Publish);
+    const { file } = args;
+    const url = this.localConfig.repository;
+
+    this.log(`file: ${file} || repo: ${url}`);
+
+    const git = createExecaCommand("git", {
+      cwd: this.localConfig.repoDir,
+    });
+
+    // check that tag exists locally
+    const { stdout } = await git(["tag"]);
+    if (!stdout.includes(file)) {
```

Comment:
Should probably be a more specific regex check, otherwise people can put nonsense values that will break this. There might also be a command to do this in git

Reply:
Used git rev-parse to check if the tag exists.